### PR TITLE
RCON websocket restarting

### PR DESCRIPTION
**Pull request by Naoki95957**
I've done some basic testing and made sure the plugin loads/unloads repeatedly and safely as well as just hitting the commands several times in a row.  There are a few special circumstances that can cause crashes and I've listed my findings. 

I've added 4 notifiers:

- `rcon_start_server` If the ws server is not listenting it'll start it up the server

- `rcon_kill_server` if the ws server is listenting, it'll instruct it to stop/close connections

- `rcon_restart_server` this combines the above 2 commands (literally) and due to some troubles with ASIO, has an overly generous wait period of 1 second. 

- `rcon_wsstatus` reports back if the endpoint is listening and if it's a server type. 

**A few things to know; I failed to get _any way of spinning/idle_ so I opted the restart to use the sleep command.**
I'm sorta new to all of this and genuinely don't know how I'd asynchronously check ASIO to properly wait in this context. Spinning caused crashes and since I every time I tried to create a thread it crashed; semaphores or more mutexes weren't really available to me.  

SO, **_if you were to call kill and start immediately, it could crash to desktop_** due to ASIO asynchronicity needing time. Where that starts and finishes is unknown to me; I read thru websocketpp a tiny bit and didn't find much about ASIO itself. 

Additionally **_if you were to change ports and fire up a new server, you MUST kill it before unloading rcon._** Starting a new server, then unloading it, causes a crash.

### CODE REVIEW

File: RCONPlugin/RCONPlugin.cpp [hunk 1]
```diff
@@ -72,12 +72,10 @@ void RCONPlugin::on_message(server* s, websocketpp::connection_hdl hdl, message_
...
 				//is_allowed
 				//Rebuild command
-				std::stringstream new_command;
-				for (auto txt : inp)
-				{
-					new_command << "\"" << replace(txt, "\"", "\\\"") << "\" ";
-				}
-				gameWrapper->Execute([cmd = new_command.str(), &_cvarManager = cvarManager, log = *logRcon](GameWrapper* gw) {
+				std::string payload = msg->get_payload();
+				//Doesn't seem neccessary? words in "" come thru in payload and sendback ok
```
Comment (Bakkes) [inline RCONPlugin/RCONPlugin.cpp:76]: I think this code exists in the case of a " being inside of quotes, like ```send_rcon(`mynotifier "my parameter has a \" in it!"`)``` should correctly get reconstructed. Have you tried the different behavior before and after making this change?

**Comment by Bakkes** [comment]
Thanks for the PR! I've added a single comment regarding some removed code but the rest looks good to me (without having tested it yet though).

The `rcon_restart_server` waiting period is fine by me, and I'll check to see if I can find a solution to `Additionally if you were to change ports and fire up a new server, you MUST kill it before unloading rcon. Starting a new server, then unloading it, causes a crash.`.

Let me know if you're able to test what I've mentioned in the comments. If not I can take a look when I've got the time

File: RCONPlugin/RCONPlugin.cpp (hunk 1 above)
Comment (Naoki95957) [reply RCONPlugin/RCONPlugin.cpp:76]: I did mess around with it and being an edge case I'm not sure of what scenarios to look out for lol. I did test it a fair amount with python and I'll share those results here:
```
			await websocket.send('sendback "te"st"')
			time.sleep(0.1)
			print(await websocket.recv())

			await websocket.send('sendback "te\\"st"')
			time.sleep(0.1)
			print(await websocket.recv())

			await websocket.send('sendback "te\\\\"st"')
			time.sleep(0.1)
			print(await websocket.recv())
``` 
printed these results:
`te st`
`te"st`
`te\"st`

If you're certain that the extra logic is needed, by all means it can be put back. It just so happened in my case that I didn't encounter that situation and just opted to pass the payload 'as-is'

**Comment by Naoki95957** [comment]
It was brought up to me by a friend that the primary issue they had with `rcon_port` was that the register cvar from the onload always had overwritten it. I know from reinstalling bakkesmod that the config.cfg contains the cvars for this plugin so is it okay and reasonable to comment/remove lines 152-155 in this PR?
```
cvarManager->registerCvar("rcon_password", "password");
cvarManager->registerCvar("rcon_port", "9002"); //Registered in the main dll now
cvarManager->registerCvar("rcon_timeout", "5");
cvarManager->registerCvar("rcon_log", "0", "Log all incoming rcon commands", true, true, 0, true, 1, true).bindTo(logRcon);
```

**Comment by Bakkes** [comment]
Only the `rcon_port` parameter is registered in the core, the others arent. Removing the rcon_port variable would actually be helpful since this would allow you to assign to the `rcon_port` cvar before the rconplugin is loaded, which will make the rcon plugin use the set port instead of the default. The others should just stay in since this is the only place they're registered.

**Comment by Naoki95957** [comment]
Got it. Having removed rcon_port only (line 153 currently): 
`cvarManager->registerCvar("rcon_port", "9002"); //Registered in the main dll now`

Seems to work just fine and allows the ws server to start up on different ports on load now :). I'll go ahead and add that commit in.

**Comment by Bakkes** [comment]
I've merged this PR in and made some changes to get rid of the mutex locking stuff (see https://github.com/bakkesmodorg/BakkesMod2-Plugins/commit/869003f79944d366bdf7fbb6b2a35c1bd2e36012)

This should also allow it to restart cleanly without `sleeps` and the rcon_*_server commands are simply wrappers around `rcon_enabled` now. Let me know if you have any comments/issues regarding the new commit!

I've tested things like:
- `rcon_port 9003; rcon_restart_server` -> works
- `rcon_stop_server; rcon_port 9004; rcon_start_server` -> works
- plugin unloading works properly
- changing ports and starting a new server -> plugin unloading without killing the server still unloads the plugin properly
